### PR TITLE
Add docker-composed based test fixture for GCS

### PR DESCRIPTION
**Pull request by tlrx**
Similarly to what has be done for Azure in #48636, this pull request adds a new `:test:fixtures:gcs-fixture` project which provides two `docker-compose` based fixture that emulates a Google Cloud Storage service.

Some code has been extracted from existing tests and placed into this new project so that it can be easily reused in other projects.

### CODE REVIEW

File: plugins/repository-gcs/qa/google-cloud-storage/build.gradle [hunk 1]
```diff
@@ -86,6 +86,19 @@ task createServiceAccountFile() {
 }
 
 task thirdPartyTest (type: Test) {
+    if (useFixture) {
+        thirdPartyTest.dependsOn createServiceAccountFile
```
Comment (tlrx) [inline plugins/repository-gcs/qa/google-cloud-storage/build.gradle]: I'm wondering if  `dependsOn tasks.postProcessFixture` is also needed?

Comment (mark-vieira) [reply plugins/repository-gcs/qa/google-cloud-storage/build.gradle]: Nope. The test fixtures plugin automatically makes all `Test` tasks depend on the fixture tasks. Essentially, when a project "uses a fixture" it's assumed that all tests require that fixture.

**Review by mark-vieira — APPROVED** [review]
LGTM 👍 

Keep 'em coming.

**Review by original-brownbear — APPROVED** [review]
LGTM2 thanks Tanguy!

File: plugins/repository-gcs/qa/google-cloud-storage/build.gradle (hunk 1 above)
Comment (tlrx) [reply plugins/repository-gcs/qa/google-cloud-storage/build.gradle]: Ok, thanks @mark-vieira !

**Comment by tlrx** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

(unrelated checkout [failure](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+packaging-tests-sample-gcp/557/))

**Comment by tlrx** [comment]
Thanks all for your reviews!

**Pull request merged**